langkit/metrics/pii.py

```diff
@@ -0,0 +1,96 @@
+from functools import partial
+from typing import Any, Dict, List, Optional
+
+import pandas as pd
+import spacy
+from presidio_analyzer import AnalyzerEngine, RecognizerResult
+from presidio_analyzer.nlp_engine import TransformersNlpEngine
+from presidio_anonymizer import AnonymizerEngine
+
+from langkit.core.metric import MultiMetric, MultiMetricResult
+
+
+def pii_presidio_metric(
+    input_name: str, language: str = "en", spacy_model: str = "en_core_web_sm", transformers_model: str = "dslim/bert-base-NER"
+) -> MultiMetric:
+    # Define which transformers model to use
+    model_config = [
+        {
+            "lang_code": language,
+            "model_name": {
+                "spacy": spacy_model,  # use a small spaCy model for lemmas, tokens etc.
+                "transformers": transformers_model,
+            },
+        }
+    ]
+    nlp_engine = TransformersNlpEngine(models=model_config)
+    analyzer = AnalyzerEngine(nlp_engine=nlp_engine)
+    anonymizer = AnonymizerEngine()
+
+    def init() -> None:
+        spacy.load(spacy_model)
+
+    def udf(text: pd.DataFrame) -> MultiMetricResult:
+        entity_types = {
+            "PHONE_NUMBER": f"{input_name}.pii.phone_number",
+            "EMAIL_ADDRESS": f"{input_name}.pii.email_address",
+            "CREDIT_CARD": f"{input_name}.pii.credit_card",
+        }
```

Comment:
We should probably include a lot more of these by default. They're all listed here

https://microsoft.github.io/presidio/supported_entities/

And maybe update pii_presidio_metric to be able to customize them too since there's a lot of country specific ones.